server/src/test/java/org/opensearch/action/admin/cluster/configuration/AddVotingConfigExclusionsRequestTests.java

```diff
@@ -450,6 +450,65 @@ public void testResolveAndCheckMaximum() {
         assertWarnings(AddVotingConfigExclusionsRequest.DEPRECATION_MESSAGE);
     }
 
+    // As of 2.0, MASTER_ROLE is deprecated to promote inclusive language.
+    // Validate node with MASTER_ROLE can be resolved by resolveVotingConfigExclusions() like before.
+    // TODO: Remove the test after removing MASTER_ROLE.
+    public void testResolveByNodeDescriptionWithDeprecatedMasterRole() {
+        final DiscoveryNode localNode = new DiscoveryNode(
+            "local",
+            "local",
+            buildNewFakeTransportAddress(),
+            emptyMap(),
+            singleton(DiscoveryNodeRole.MASTER_ROLE),
+            Version.CURRENT
+        );
+        final VotingConfigExclusion localNodeExclusion = new VotingConfigExclusion(localNode);
+
+        final ClusterState clusterState = ClusterState.builder(new ClusterName("cluster"))
+            .nodes(new Builder().add(localNode).localNodeId(localNode.getId()))
+            .build();
+
+        assertThat(makeRequestWithNodeDescriptions("_local").resolveVotingConfigExclusions(clusterState), contains(localNodeExclusion));
+        assertWarnings(AddVotingConfigExclusionsRequest.DEPRECATION_MESSAGE);
```

Comment:
This assertion seems unnecessary for this test case - you can probably remove it

Reply:
👏 You are so careful that notice such detail! Good catch, I will try to replace with `exceptWarnings()`, removing is not feasible, because any warnings can fail the test.